src/components/GalleryEditor/GalleryEditorContext.tsx

```diff
@@ -22,63 +58,420 @@ type GalleryEditorProviderProps = PropsWithChildren<{
+          name: collection.name,
+          tokens: tokens.map((token) => token.id),
+        };
+      };
+
+      const localCollectionToCreatedCollection = (
+        collection: CollectionState
+      ): CreateCollectionInGalleryInput => {
+        const tokens = Object.values(collection.sections).flatMap((section) =>
+          section.items.filter((item) => item.kind === 'token')
+        );
+
+        const layout = generateLayoutFromCollectionNew(collection.sections);
+
+        return {
+          givenID: collection.dbid,
+          hidden: collection.hidden,
+          collectorsNote: collection.collectorsNote,
+          tokenSettings: tokens.map((token) => {
+            return { tokenId: token.id, renderLive: collection.liveDisplayTokenIds.has(token.id) };
+          }),
+          layout,
+          name: collection.name,
+          tokens: tokens.map((token) => token.id),
+        };
+      };
+
+      const updatedCollections: UpdateCollectionInput[] = Object.values(collections)
+        .filter((collection) => !collection.localOnly)
+        .map(localCollectionToUpdatedCollection);
+
+      const createdCollections: CreateCollectionInGalleryInput[] = Object.values(collections)
+        .filter((collection) => collection.localOnly)
+        .map(localCollectionToCreatedCollection);
+
+      const deletedCollections = [...deletedCollectionIds];
+
+      const order = [...Object.values(collections).map((collection) => collection.dbid)];
+
+      try {
+        await save({
+          variables: {
+            input: {
+              galleryId,
+
+              name,
+              description,
+              caption,
+
+              order,
+
+              createdCollections,
+              updatedCollections,
+              deletedCollections,
+            },
+          },
+        });
+
+        // Make sure we reset our "Has unsaved changes comparison point"
+        setInitialName(name);
+        setInitialDescription(description);
+        setInitialCollections(collections);
+      } catch (error) {
+        pushToast({
+          autoClose: false,
+          message: "Something went wrong while saving your gallery. We're looking into it.",
+        });
+
+        if (error instanceof Error) {
+          reportError(error, { tags: { galleryId } });
+        } else {
+          reportError(
+            new ErrorWithSentryMetadata('Something unexpected went wrong while saving a gallery', {
+              galleryId,
+            })
+          );
+        }
+      }
+    },
+    [collections, deletedCollectionIds, description, name, query.galleryById, reportError, save]
+  );
+
+  const [initialName, setInitialName] = useState(name);
+  const [initialDescription, setInitialDescription] = useState(description);
+  const [initialCollections, setInitialCollections] = useState(collections);
+  const hasUnsavedChanges = useMemo(() => {
+    function removeIdsFromCollections(collections: CollectionState[]): CollectionState[] {
```

Comment:
could you explain the reason for this

Reply:
turns out we don't need it now. thought i needed it because the ids get changed by the server. but we manually keep track of our save point anyway so we good